src/NServiceBus.PersistenceTesting/IPersistenceTestsConfiguration.cs

```diff
@@ -41,21 +41,23 @@ public interface IPersistenceTestsConfiguration
 
         Task Cleanup();
 
-        Func<ContextBag> GetContextBagForSagaStorage { get; } //TODO why is this not used?
+        Func<ContextBag> GetContextBagForSagaStorage { get; }
         Func<ContextBag> GetContextBagForOutbox { get; }
     }
 
     // Consumers of this source package have to implement the remaining properties via partial class to configure the tests infrastructure.
     public partial class PersistenceTestsConfiguration : IPersistenceTestsConfiguration
     {
-        public PersistenceTestsConfiguration(TimeSpan? sessionTimeout = null)
+        public PersistenceTestsConfiguration(TestVariant variant, TimeSpan? sessionTimeout = null)
         {
             SessionTimeout = sessionTimeout;
+            Variant = variant;
         }
 
         public Func<ContextBag> GetContextBagForSagaStorage { get; private set; } = () => new ContextBag();
         public Func<ContextBag> GetContextBagForOutbox { get; private set; } = () => new ContextBag();
         public TimeSpan? SessionTimeout { get; }
+        public TestVariant Variant { get; }
```

Comment:
this has been added to `PersistenceTestsConfiguration` directly instead to `IPersistenceTestsConfiguration`. I think this is okay in this case as the interface is mostly here as a mean to enforce implementation of specific properties/methods in downstream since this is already on the base class this isn't necessary and the tests themselves shouldn't require access to the `TestVariant`, so it should be correct this way.